nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -356,6 +361,107 @@ public void testListTables() {
     Assertions.assertThat(catalog.tableExists(TABLE_IDENTIFIER)).isTrue();
   }
 
+  private void testRegister(TableIdentifier identifier, List<String> metadataVersionFiles) {
+    Assertions.assertThat(catalog.dropTable(identifier, false)).isFalse();
+    Assertions.assertThat(catalog.registerTable(
+            identifier,
+            "file:" + metadataVersionFiles.get(0))).isNotNull();
+    Table newTable = catalog.loadTable(identifier);
+    Assertions.assertThat(newTable).isNotNull();
+    TableOperations ops = ((HasTableOperations) newTable).operations();
+    String metadataLocation = ((NessieTableOperations) ops).currentMetadataLocation();
+    Assertions.assertThat("file:" + metadataVersionFiles.get(0)).isEqualTo(metadataLocation);
+    Assertions.assertThat(catalog.tableExists(identifier)).isTrue();
```

Comment:
load table is called above. So, this check is not needed as both are same functionality.